core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/search/TermFilter.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2020-2022 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.request.search;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import com.b2international.index.query.Expression;
+import com.google.common.collect.Iterables;
+
+/**
+ * Represents a {@link Serializable} encapsulation of term search configuration.
+ * 
+ * @since 7.11
+ */
+public abstract class TermFilter implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+	public static MatchTermFilter.Builder match() {
+		return new MatchTermFilter.Builder();
+	}
+	
+	public static ExactTermFilter.Builder exact() {
+		return new ExactTermFilter.Builder();
+	}
+	
+	public static ParsedTermFilter.Builder parsed() {
+		return new ParsedTermFilter.Builder();
+	}
+	
+	public static MoreLikeThisTermFilter.Builder mlt() {
+		return new MoreLikeThisTermFilter.Builder();
+	}
+
+	/**
+	 * Get the search term(s) from this term filter. Certain implementations only support a single term to be defined.
+	 * 
+	 * @return a collection of search terms, never <code>null</code>
+	 */
+	public abstract Set<String> getTerms();
+	
+	public final String getSingleTermOrNull() {
+		final Set<String> terms = getTerms();
+		return terms.size() == 1 ? Iterables.getOnlyElement(terms) : null;
+	}
+	
+	public Expression toExpression(String field) {
+		return toExpression(field, "text", "exact", "prefix");
+	}
+	
+	public abstract Expression toExpression(String field, String textFieldSuffix, String exactFieldSuffix, String prefixFieldSuffix);
+	
+	protected final String fieldAlias(String field, String textFieldSuffix) {
+		return String.join(".", field, textFieldSuffix);
+	}
```

Comment:
Second argument name should be more general:

```suggestion
	protected final String fieldAlias(String field, String fieldSuffix) {
		return String.join(".", field, fieldSuffix);
	}
```